AutoNode/validator.py

```diff
@@ -0,0 +1,170 @@
+    bls_key_dir,
+    node_config,
+    validator_config,
+    saved_wallet_pass_path,
+    harmony_dir,
+)
+from .blockchain import (
+    get_latest_header,
+    get_latest_headers,
+    get_staking_epoch,
+    get_current_epoch,
+    get_all_validator_addresses,
+    get_validator_information
+)
+from .node import (
+    wait_for_node_response,
+    assert_no_bad_blocks
+)
+from .util import (
+    check_min_bal_on_s0,
+    input_with_print
+)
+
+log_path = f"{harmony_dir}/autonode_validator.log"
+
+
+def _add_bls_key_to_validator():
+    wait_for_node_response(node_config['endpoint'], verbose=True)
+    chain_val_info = get_validator_information(validator_config['validator-addr'], node_config['endpoint'])
+    bls_keys = set(x.replace('0x', '') for x in chain_val_info["validator"]["bls-public-keys"])
+    for k in (x.replace('0x', '') for x in node_config["public-bls-keys"]):
+        if k not in bls_keys:  # Add imported BLS key to existing validator if needed
+            print(f"{Typgpy.OKBLUE}Adding bls key: {Typgpy.OKGREEN}{k}{Typgpy.OKBLUE} "
+                  f"to validator: {Typgpy.OKGREEN}{validator_config['validator-addr']}{Typgpy.ENDC}")
+            try:
+                wait_for_node_response(node_config['endpoint'], verbose=True)
+                response = cli.single_call(f"hmy --node={node_config['endpoint']} staking edit-validator "
+                                           f"--validator-addr {validator_config['validator-addr']} "
+                                           f"--add-bls-key {k} --passphrase-file {saved_wallet_pass_path} "
+                                           f"--bls-pubkeys-dir {bls_key_dir}")
+                print(f"\n{Typgpy.OKBLUE}Edit-validator transaction response: "
+                      f"{Typgpy.OKGREEN}{response}{Typgpy.ENDC}")
+            except (json.decoder.JSONDecodeError, RuntimeError) as e:
+                print(f"{Typgpy.FAIL}Failed to edit validator!\n\tError: {e}{Typgpy.ENDC}")
+        else:
+            print(f"{Typgpy.WARNING}Bls key: {Typgpy.OKGREEN}{k}{Typgpy.WARNING} "
+                  f"is already present, ignoring...{Typgpy.ENDC}")
+    _verify_node_sync()
+
+
+def _verify_node_sync():
+    print(f"{Typgpy.OKBLUE}Verifying Node Sync...{Typgpy.ENDC}")
+    wait_for_node_response(node_config['endpoint'], verbose=True)
+    wait_for_node_response("http://localhost:9500/", verbose=True)
+    curr_headers = get_latest_headers("http://localhost:9500/")
+    curr_epoch_shard = curr_headers['shard-chain-header']['epoch']
+    curr_epoch_beacon = curr_headers['beacon-chain-header']['epoch']
+    ref_epoch = get_latest_header(node_config['endpoint'])['epoch']
+    has_looped = False
+    while curr_epoch_shard != ref_epoch or curr_epoch_beacon != ref_epoch:
+        sys.stdout.write(f"\rWaiting for node to sync: shard epoch ({curr_epoch_shard}/{ref_epoch}) "
+                         f"& beacon epoch ({curr_epoch_beacon}/{ref_epoch})")
+        sys.stdout.flush()
```

Comment:
what is wrong with plain print

Reply:
would spam the screen, think this is cleaner :)